pkg/storage/utils/localfs/localfs.go

```diff
@@ -1147,14 +1147,14 @@ func (fs *localfs) RestoreRecycleItem(ctx context.Context, restoreKey string) er
 func (fs *localfs) propagate(ctx context.Context, leafPath string) error {
 
 	var root string
-	if fs.isShareFolderChild(ctx, leafPath) {
+	if fs.isShareFolderChild(ctx, leafPath) || strings.HasSuffix(path.Clean(leafPath), fs.conf.ShareFolder) {
 		root = fs.wrapReferences(ctx, "/")
```

Comment:
@ishank011 what is the meaning of this?

Reply:
This takes care of the corner case when the leaf path is the shares directory itself. Since it isn't a share child, the root we assign to it is the data directory which fails since that is not a prefix of the references directory.